plugins/lighthouse/src/components/AuditList/AuditListTable.tsx

```diff
@@ -13,81 +13,107 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import React, { FC, useMemo } from 'react';
-import {
-  Table,
-  TableBody,
-  TableCell,
-  TableContainer,
-  TableHead,
-  TableRow,
-} from '@material-ui/core';
-import { makeStyles } from '@material-ui/core/styles';
-import { Website } from '../../api';
+import React, { FC, useState } from 'react';
+import { Table, TableColumn, TrendLine, useApi } from '@backstage/core';
+import { Website, lighthouseApiRef } from '../../api';
+import { useInterval } from 'react-use';
 import {
+  formatTime,
   CATEGORIES,
   CATEGORY_LABELS,
-  SparklinesDataByCategory,
   buildSparklinesDataForItem,
 } from '../../utils';
-import Audit from '../Audit';
-
-const useStyles = makeStyles(theme => ({
-  table: {
-    minWidth: 650,
-  },
-  status: {
-    textTransform: 'capitalize',
-  },
-  link: {
-    paddingTop: theme.spacing(2),
-    paddingBottom: theme.spacing(2),
-    display: 'inline-block',
-  },
-  statusCell: { whiteSpace: 'nowrap' },
-  sparklinesCell: { minWidth: 120 },
-}));
+import { Link } from '@material-ui/core';
+import AuditStatusIcon from '../AuditStatusIcon';
 
 export const AuditListTable: FC<{ items: Website[] }> = ({ items }) => {
-  const classes = useStyles();
-  const categorySparklines: Record<string, SparklinesDataByCategory> = useMemo(
-    () =>
-      items.reduce(
-        (res, item) => ({
-          ...res,
-          [item.url]: buildSparklinesDataForItem(item),
-        }),
-        {},
+  const [websiteState, setWebsiteState] = useState(items);
+  const lighthouseApi = useApi(lighthouseApiRef);
+
+  const runRefresh = async (website: Website) => {
+    const response = await lighthouseApi.getWebsiteForAuditId(
+      website.lastAudit.id,
+    );
+    const auditStatus = response.lastAudit.status;
+    if (auditStatus === 'COMPLETED' || auditStatus === 'FAILED') {
+      const newWebsiteData = websiteState.slice(0);
+      newWebsiteData[
+        newWebsiteData.findIndex((w) => w.url === response.url)
+      ] = response;
+      setWebsiteState(newWebsiteData);
+    }
+  };
+
+  websiteState.forEach((website) => {
+    useInterval(
```

Comment:
Can't call this hook in a loop. I think it should be possible to invert this a bit and have a single `useInterval` that updates all websites if needed. And having that run only if there is at least one website in status RUNNING.

Reply:
Yea, i was a bit sceptical of that myself. Pushed a new solution to this. Hopefully it's a bit better.

Reply:
No you didn't? x)

Reply:
Wops. Fixed. 😳